src/frontend-samples/version-compare-sample/VersionCompareClient.ts

```diff
@@ -0,0 +1,162 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { assert } from "@bentley/bentleyjs-core";
+import { AuthorizedFrontendRequestContext, IModelApp, IModelConnection } from "@bentley/imodeljs-frontend";
+import { IncludePrefix, request, Response } from "@bentley/itwin-client";
+import { AuthorizationClient } from "@itwinjs-sandbox";
+import { VersionCompareApi } from "./VersionCompareApi";
+
+interface ProjectContext {
+  projectId: string;
+  iModelId: string;
+  requestContext: AuthorizedFrontendRequestContext;
+}
+
+export class VersionCompareClient {
+
+  // default initialization, will be replaced when the app loads.
+  private static _projectContext: ProjectContext = {
+    projectId: "",
+    iModelId: "",
+    requestContext: null as any,
+  };
+
+  /** Creates a context with all API request will be made in. */
+  public static async populateContext(iModel: IModelConnection) {
+    // Check if already populated
+    if (VersionCompareClient._projectContext.requestContext !== null) return;
+    const requestContext = await VersionCompareApi.getRequestContext();
+    const projectId = iModel.contextId;
+    const iModelId = iModel.iModelId;
+    assert(projectId !== undefined);
+    assert(iModelId !== undefined);
+    VersionCompareClient._projectContext = { projectId, iModelId, requestContext };
+  }
+
+  /** Gets the changes in version using REST API.  Will response with JSON describing changes made between the two changesets.  Pass the same changeset Id as the start and end to view the changes for that changeset.
+   * Read more at {@link https://developer.bentley.com/api-groups/project-delivery/apis/changed-elements/operations/get-comparison/|developer.bentley.com}
+   * @param startChangesetId The oldest changeset id to use for the comparison.
+   * @param endChangesetId The newest changeset id to use for the comparison.
+   * @return A OK response will be formatted as follows:
+   * ```
+   * {
+      "changedElements": {
+        "elements": ["0x30000000f69"],
+        "classIds": ["0x670"],
+        "opcodes": [23],
+        "modelIds": ["0x20000000002"],
+        "type": [1],
+        "properties": [
+          ["UserLabel"]
+        ],
+        "oldChecksums": [
+          [1448094486]
+        ],
+        "newChecksums": [
+          [362149254]
+        ],
+        "parentIds": ["0"],
+        "parentClassIds": ["0"]
+      }
+    }
+   * ```
+   */
+  public static async getChangedElements(startChangesetId: string, endChangesetId: string): Promise<any | undefined> {
```

Comment:
We may want to cast the result to [ChangedElements](https://www.itwinjs.org/reference/imodeljs-common/entities/changedelements/)

Reply:
Casted results in the compareChangesets method in ChangedElementApi.ts